src/lib.rs

```diff
@@ -507,6 +513,23 @@ impl RestClient {
             Ok(Either::A((got, _))) => Ok(got),
             Ok(Either::B((_, _))) => Err(Error::TimeoutError),
             Err(_) => Err(Error::RequestError),
+        let work = req.select2(timeout).then(|res| {
```

Comment:
The new req.select2 logic seems to be inserted inside the old one instead of replacing it. This produces compile errors.